fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java

```diff
@@ -100,18 +100,38 @@ public FedoraBinaryImpl(final Node node) {
 
     @Override
     public FedoraResource getDescription() {
+        final Node descNode = getDescriptionNode();
+        if (descNode == null) {
+            return null;
+        }
         return new NonRdfSourceDescriptionImpl(getDescriptionNode());
     }
 
     @Override
     protected Node getDescriptionNode() {
         try {
+            final Node node = getNode();
+            if (isMemento()) {
+                final String mementoName = node.getName();
+                return node.getNode("../../" + FEDORA_DESCRIPTION + "/" + LDPCV_TIME_MAP + "/" + mementoName);
+            }
             return getNode().getNode(FEDORA_DESCRIPTION);
         } catch (final RepositoryException e) {
             throw new RepositoryRuntimeException(e);
         }
     }
 
+    private Node getDescriptionNodeOrNull() {
```

Comment:
Why would there not be a description node? It is created as part of the Binary, right?

Reply:
For the annoying period between creating the historic binary memento and importing the description for it. Rather than creating a description and allowing the user to update it, this PR simply doesn't create it until the client provides it.

For mementos created from the original binary resource, the description is automatically created.